### ISSUE TITLE
Cannot destructure property `activity` of 'undefined' or 'null' when running develop

### ISSUE THREAD

**Issue by kmcaloon** [gatsbyjs/gatsby#18091]
Hi all, so I am running a pretty fresh install of Gatby and during develop I am getting the following error after the server successfully starts: 

```
 ERROR 

UNHANDLED REJECTION Cannot destructure property `activity` of 'undefined' or 'null'.



  TypeError: Cannot destructure property `activity` of 'undefined' or 'null'.
  
  - develop.js:126 createIndexHtml
    [dartdrones.com]/[gatsby]/dist/commands/develop.js:126:27
  
  - develop.js:338 FSWatcher.chokidar.watch.on
    [dartdrones.com]/[gatsby]/dist/commands/develop.js:338:11
  
  - index.js:483 FSWatcher.emitWithAll
    [dartdrones.com]/[chokidar]/index.js:483:8
  
  - index.js:573 FSWatcher._emit
    [dartdrones.com]/[chokidar]/index.js:573:10
  
  - fsevents-handler.js:250 FsEventsHandler.handleEvent
    [dartdrones.com]/[chokidar]/lib/fsevents-handler.js:250:14
  
  - fsevents-handler.js:205 FsEventsHandler.addOrChange
    [dartdrones.com]/[chokidar]/lib/fsevents-handler.js:205:8
  
  - fsevents-handler.js:303 watchCallback
    [dartdrones.com]/[chokidar]/lib/fsevents-handler.js:303:21
  
  - fsevents-handler.js:103 filteredListener
    [dartdrones.com]/[chokidar]/lib/fsevents-handler.js:103:7
  
  - fsevents-handler.js:127 cont.listeners.forEach.list
    [dartdrones.com]/[chokidar]/lib/fsevents-handler.js:127:11
  
  - Set.forEach
  
  - fsevents-handler.js:126 cont.watcher.createFSEventsInstance
    [dartdrones.com]/[chokidar]/lib/fsevents-handler.js:126:24
  
```

Not really sure where to start here?

**Comment by davwheat** [comment]
I'm getting the same error with a custom `html.js` in my `src` folder.

**Comment by gc** [comment]
^ Same here.

Error comes from here: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/commands/develop.js#L74

**Comment by n0k5** [comment]
Same ^ extremely annoying as it crashes every time I edit the html.js file

**Comment by manuelnas** [comment]
Is there any kind of fix or known reason for this error? I keep getting it, and while it's not breaking because it doesn't happen in build, it's incredibly frustrating. I too have a custom `HTML.js` file in my `src` folder, but this happens when I don't change anything in it at seemingly random times.

**Comment by karataev** [comment]
In my case this error happens every time I edit `gatsby-ssr.js` during development.

**Comment by sidharthachatterjee** [comment]
Apologies for this! Can you please link to a [minimal reproduction](https://www.gatsbyjs.org/contributing/how-to-make-a-reproducible-test-case/)?

We'll get this fixed right after. 

Also, please upgrade to the latest `gatsby` and `gatsby-cli` before that. We might've fixed it already 🙂

**Comment by davwheat** [comment]
@sidharthachatterjee 

It's tough to create a repoduction. 

@karataev says it only happens with an edited `gatsby-ssr.js` but I don't have an edited ssr file.

For me, it happened after I added a custom html.js file but it only happens sometimes. Generally, closing my IDE and re-running `gatsby develop` fixes it until it next happens.

**Comment by karataev** [comment]
> Apologies for this! Can you please link to a [minimal reproduction](https://www.gatsbyjs.org/contributing/how-to-make-a-reproducible-test-case/)?
> 
> We'll get this fixed right after.
> 
> Also, please upgrade to the latest `gatsby` and `gatsby-cli` before that. We might've fixed it already 🙂

For my case steps to reproduce are simple:
1. Start a new project with the default starter: `gatsby new my-default-starter https://github.com/gatsbyjs/gatsby-starter-default`
2. `gatsby develop`
3. Do any change to `gatsby-ssr.js` (add `console.log`, for example) and save the file.

Catch the error in the console:
```
 ERROR

UNHANDLED REJECTION Cannot destructure property `activity` of 'undefined' or 'null'.



  TypeError: Cannot destructure property `activity` of 'undefined' or 'null'.

  - develop.js:131 createIndexHtml
    [gatsby-test]/[gatsby]/dist/commands/develop.js:131:27

  - develop.js:352 FSWatcher.chokidar.watch.on
    [gatsby-test]/[gatsby]/dist/commands/develop.js:352:11

  - index.js:494 FSWatcher.emitWithAll
    [gatsby-test]/[gatsby]/[chokidar]/index.js:494:8

  - index.js:585 FSWatcher._emit
    [gatsby-test]/[gatsby]/[chokidar]/index.js:585:10

  - nodefs-handler.js:360 closer._watchWithNodeFs
    [gatsby-test]/[gatsby]/[chokidar]/lib/nodefs-handler.js:360:20
```

I updated `gatsby` and `gatsby-cli` to the latest versions.
```
Gatsby CLI version: 2.8.5
Gatsby version: 2.17.4
```

`gatsby-info`
```
  System:
    OS: Windows 10
    CPU: (4) x64 Intel(R) Core(TM) i3-7100 CPU @ 3.90GHz
  Binaries:
    Yarn: 1.3.2 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 2.7.15 - /c/Python27/python
  Browsers:
    Edge: 44.18362.387.0
  npmPackages:
    gatsby: ^2.17.4 => 2.17.4
    gatsby-image: ^2.2.29 => 2.2.29
    gatsby-plugin-manifest: ^2.2.23 => 2.2.23
    gatsby-plugin-offline: ^3.0.16 => 3.0.16
    gatsby-plugin-react-helmet: ^3.1.13 => 3.1.13
    gatsby-plugin-sharp: ^2.2.32 => 2.2.32
    gatsby-source-filesystem: ^2.1.33 => 2.1.33
    gatsby-transformer-sharp: ^2.3.0 => 2.3.0
```

**Comment by huy-nguyen** [comment]
Just want to contribute my findings to this issue. I've run into it a few times and always as a result of on-the-fly edits to `html.js` while running `gatsby develop`. Stopping and restarting `gatsby develop` fixes it.

**Comment by ErisDS** [comment]
Seeing the exact same problem as huy-nguyen - any edits to a custom `html.js` whilst running `gatsby develop` cause this error.

**Comment by sidharthachatterjee** [comment]
This was fixed by @blainekasten in https://github.com/gatsbyjs/gatsby/pull/19544 and published in `gatsby@2.17.15` 🙂 

Please reopen in case you're still seeing the issue after upgrading!

**Comment by MsMaddyMac** [comment]
I am running into this error when running gatsby develop. The current version of the CLI is 2.16.1 and gatsby version is 2.18.7. Please let me know what to do here.
